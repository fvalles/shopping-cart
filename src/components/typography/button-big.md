Basic button big text component with default color="primary":

```jsx
<ButtonBig>Lorem ipsum</ButtonBig>
```

Button big text component with color="summaryText":

```jsx
<ButtonBig color="summaryText">Lorem ipsum</ButtonBig>
```
