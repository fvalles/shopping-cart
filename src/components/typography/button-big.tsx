import styled from 'styled-components';
import { TypographyProps } from './common';

export const ButtonBig = styled.span<TypographyProps>`
  color: ${({ color = 'primary', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 25px;
`;
