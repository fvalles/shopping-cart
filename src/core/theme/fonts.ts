export const Fonts = {
  light: 'Avenir Light',
  medium: 'Avenir Medium',
  regular: 'Avenir Regular',
};

export type FontsType = typeof Fonts;
