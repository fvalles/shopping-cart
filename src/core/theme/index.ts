export { Colors } from './colors';
export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
