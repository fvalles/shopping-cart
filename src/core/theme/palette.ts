export const palette = {
  // color names are based on https://colornamer.netlify.app/
  black: '#000000',
  blackMarket: '#212240',
  blackMarketLight: '#21224029',
  bleachedSilk: '#F3F3F3',
  faintFern: '#DBDBE0',
  mischka: '#A6A7B3',
  sharpGrey: '#CACAD1',
  stormGrey: '#717285',
  venetianNights: '#7350FF',
  white: '#FFFFFF',
};
