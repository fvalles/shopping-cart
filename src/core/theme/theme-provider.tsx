import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Colors, Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Colors,
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
