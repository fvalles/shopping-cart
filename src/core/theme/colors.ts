import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  primary: palette.venetianNights,
  productImageBorder: palette.sharpGrey,
  productPrice: palette.black,
  productQuantity: palette.faintFern,
  productTitle: palette.mischka,
  summaryBackground: palette.bleachedSilk,
  summaryHeading: palette.stormGrey,
  summaryText: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
