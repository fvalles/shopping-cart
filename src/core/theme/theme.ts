import { DefaultTheme } from 'styled-components';
import { Colors } from './colors';

const theme: DefaultTheme = {
  Colors,
};

export { theme };
