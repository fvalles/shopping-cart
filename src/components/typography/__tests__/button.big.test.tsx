import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ButtonBig } from '../button-big';

describe('button big component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <ButtonBig>Lorem ipsum</ButtonBig>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#7350FF');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Regular,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '20px');
    expect(result).toHaveStyleRule('font-weight', 'bold');
    expect(result).toHaveStyleRule('line-height', '25px');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom summaryText color', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <ButtonBig color="summaryText">Lorem ipsum</ButtonBig>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#212240');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Regular,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '20px');
    expect(result).toHaveStyleRule('font-weight', 'bold');
    expect(result).toHaveStyleRule('line-height', '25px');
    expect(result).toMatchSnapshot();
  });
});
