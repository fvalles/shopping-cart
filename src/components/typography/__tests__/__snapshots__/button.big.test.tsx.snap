// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button big component should render with custom summaryText color 1`] = `
.c0 {
  color: #212240;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 25px;
}

<span
  className="c0"
  color="summaryText"
>
  Lorem ipsum
</span>
`;

exports[`button big component should render with default css properties 1`] = `
.c0 {
  color: #7350FF;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 25px;
}

<span
  className="c0"
>
  Lorem ipsum
</span>
`;
